disnake/ext/commands/common_bot_base.py

```diff
@@ -635,7 +635,7 @@ async def _watchdog(self):
             extensions = set()
             for name, module in self.extensions.items():
                 file = module.__file__
-                if os.stat(file).st_mtime > last:
+                if file and os.stat(file).st_mtime > last:
```

Comment:
perhaps we should raise a warning or log if file is falsey.

Reply:
That could get fairly annoying, since it would log at least once every second. The condition was mostly changed to make pyright shut up, as `__file__` should only be `None` for builtin modules

Reply:
As my time with importlib has shown me, a module does not require a file containing the code exists. I believe there is an attribute on the spec of a module which we could convert this to use in the future.


That aside, it's something to consider for the future. Looks good for now.

Reply:
Ah, fair enough.
Either way, reload is only meant to work with modules that have associated files, since the modification time needs to be checked in some way.